src/NzbDrone.Core/Parser/ParsingService.cs

```diff
@@ -354,91 +342,113 @@ private ParsedEpisodeInfo ParseSpecialEpisodeTitle(string title, Series series)
+        {
+            Func<Movie, bool> isNotNull = movie => movie != null;
+            if (parsedMovieInfo.Year > 1800)
+            {
+                movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle, parsedMovieInfo.Year);
+                if (isNotNull(movieByTitleAndOrYear))
+                {
+                    return true;
                 }
+            }
+            movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
+            if (isNotNull(movieByTitleAndOrYear))
+            {
+                return true;
+            }
+            movieByTitleAndOrYear = null;
+            return false;
+        }
 
-				if (possibleMovie != null && (parsedMovieInfo.Year < 1800 || possibleMovie.Year == parsedMovieInfo.Year))
-				{
-					return possibleMovie;
-				}
+        private static bool TryGetMovieBySearchCriteria(ParsedMovieInfo parsedMovieInfo, SearchCriteriaBase searchCriteria, out Movie possibleMovie)
+        {
+            possibleMovie = null;
+            var possibleTitles = new List<string>();
+            Movie possibleMovieBySearchCriteria = null;
 
-                    
-            }
+            possibleTitles.Add(searchCriteria.Movie.CleanTitle);
+            var arabicRomanNumeralMappings = RomanNumeralParser.GetArabicRomanNumeralsMapping();
 
-            Movie movie = null;
+            foreach (string altTitle in searchCriteria.Movie.AlternativeTitles)
+            {
+                possibleTitles.Add(altTitle.CleanSeriesTitle());
+            }
 
-            if (searchCriteria == null)
+            foreach (string title in possibleTitles)
             {
-                if (parsedMovieInfo.Year > 1800)
-                {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle, parsedMovieInfo.Year);
-                }
-                else
+                if (title == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
                 {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
+                    possibleMovieBySearchCriteria = searchCriteria.Movie;
                 }
 
-                if (movie == null)
+                foreach (ArabicRomanNumeral numeralMapping in arabicRomanNumeralMappings)
                 {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
-                }
-                // return movie;
-            }
-
+                    string arabicNumeral = numeralMapping.ArabicNumeralAsString;
+                    string romanNumeral = numeralMapping.RomanNumeralLowerCase;
 
+                    if (title.Replace(arabicNumeral, romanNumeral) == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
+                    {
+                        possibleMovieBySearchCriteria = searchCriteria.Movie;
+                    }
 
-            if (movie == null && imdbId.IsNotNullOrWhiteSpace())
```

Comment:
Is there a reason why trying to get a movie in case a `imdbId` exists, is done at the very end. Wouldn't it be much more effective to try getting a movie with an existing `imdbId` right away because wouldn't the likelihood of identifying a movie that way be much higher than with any given title and/or year and/or `SearchCriteria' ? I am asking this from the scope of the method w/o taking into account how it might be called. For instance it is always called in a way that either an `imdbId` or `searchCriteria` parameter is filled but not both, which to me would be reason enough to split this method up such that it reflects the way it is actually always called in reality.